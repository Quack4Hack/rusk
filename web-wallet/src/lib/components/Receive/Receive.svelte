<svelte:options immutable={true} />

<script>
  import { mdiArrowLeft, mdiContentCopy } from "@mdi/js";
  import { onMount } from "svelte";

  import { AppAnchorButton } from "$lib/components";
  import { Button, QrCode } from "$lib/dusk/components";
  import { toast } from "$lib/dusk/components/Toast/store";

  /** @type {string} */
  export let address = "";

  /** @type {boolean} */
  export let hideBackButton = false;

  /** @type {HTMLElement} */
  let figureElement;

  let qrWidth = 0;

  onMount(() => {
    const resizeObserver = new ResizeObserver((entries) => {
      qrWidth = entries[0].contentRect.width;
    });

    resizeObserver.observe(figureElement);

    return () => resizeObserver.disconnect();
  });
</script>

<div class="receive">
  <figure class="receive__address-qr-figure" bind:this={figureElement}>
    <QrCode value={address} className="receive__qr" width={qrWidth} />

    <figcaption class="receive__address">
      <samp>{address}</samp>
    </figcaption>
  </figure>

  <div class="receive__buttons">
    {#if !hideBackButton}
      <AppAnchorButton
        className="receive__button"
        icon={{ path: mdiArrowLeft }}
        href="/dashboard"
        text="Back"
        variant="tertiary"
      />
    {/if}
    <Button
      className="receive__button"
      icon={{ path: mdiContentCopy }}
      on:click={() => {
        navigator.clipboard.writeText(address);
        toast("success", "Address copied", mdiContentCopy);
      }}
      text="Copy"
    />
  </div>
</div>

<style lang="postcss">
  .receive {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    gap: var(--default-gap);
    z-index: 3;

    &__address-qr-figure {
      display: flex;
      flex-direction: column;
      gap: var(--default-gap);
      align-items: center;
    }

    &__address,
    :global(&__qr) {
      border-radius: 1.5em;
      width: 100%;
    }

    &__address {
      line-break: anywhere;
      padding: 0.75em 1em;
      background-color: transparent;
      border: 1px solid var(--primary-color);
    }

    :global(&__qr) {
      padding: 0.625em;
      background-color: var(--background-color-alt);
    }

    :global(&__button) {
      flex: 1;
    }

    &__buttons {
      display: flex;
      flex-direction: row;
      gap: var(--default-gap);
      width: 100%;
    }
  }
</style>
