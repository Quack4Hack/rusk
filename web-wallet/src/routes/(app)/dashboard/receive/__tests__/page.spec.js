import { afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import Receive from "../+page.svelte";

vi.useFakeTimers();

describe("Receive", () => {
  afterEach(cleanup);

  it("should render the receive page", async () => {
    const { container } = render(Receive);

    await vi.advanceTimersToNextTimerAsync();

    expect(container.firstChild).toMatchSnapshot();
  });
});
