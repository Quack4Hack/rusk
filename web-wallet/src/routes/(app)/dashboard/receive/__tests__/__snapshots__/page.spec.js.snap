// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Receive > should render the receive page 1`] = `
<div
  class="dusk-card dusk-card--gap-medium"
>
  <div
    class="dusk-card__header-container"
  >
    <header
      class="dusk-card__header"
      slot="header"
    >
      <div
        class="dusk-card__heading-container"
      >
        <h3
          class="h4"
        >
          Receive
        </h3>
         
        <div
          class="dusk-card__heading-icons"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.07 19L9.24 16.18L19 6.42L17.58 5L7.82 14.76L5 11.94V19Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,1L3,5V11C3,16.55 6.84,21.74 12,23C17.16,21.74 21,16.55 21,11V5L12,1M12,7C13.4,7 14.8,8.1 14.8,9.5V11C15.4,11 16,11.6 16,12.3V15.8C16,16.4 15.4,17 14.7,17H9.2C8.6,17 8,16.4 8,15.7V12.2C8,11.6 8.6,11 9.2,11V9.5C9.2,8.1 10.6,7 12,7M12,8.2C11.2,8.2 10.5,8.7 10.5,9.5V11H13.5V9.5C13.5,8.7 12.8,8.2 12,8.2Z"
            />
          </svg>
          
        </div>
      </div>
    </header>
  </div>
   
  <div
    class="dusk-card__body-container"
  >
    <div
      class="dusk-exclusive-choice"
      role="radiogroup"
    >
      <input
        class="dusk-exclusive-choice__radio"
        id="some-generated-id-shielded"
        name="some-generated-id"
        type="radio"
        value="shielded"
      />
       
      <label
        class="dusk-exclusive-choice__label"
        for="some-generated-id-shielded"
      >
        Shielded
      </label>
      <input
        class="dusk-exclusive-choice__radio"
        id="some-generated-id-unshielded"
        name="some-generated-id"
        type="radio"
        value="unshielded"
      />
       
      <label
        class="dusk-exclusive-choice__label"
        for="some-generated-id-unshielded"
      >
        Unshielded
      </label>
    </div>
     
    <div
      class="receive svelte-14g7par"
    >
      <figure
        class="receive__address-qr-figure svelte-14g7par"
      >
        <p>
          Unable to get QR code
        </p>
         
        <p>
          Error: No input text
        </p>
        
         
        <figcaption
          class="receive__address svelte-14g7par"
        >
          <samp>
            
          </samp>
        </figcaption>
      </figure>
       
      <div
        class="receive__buttons svelte-14g7par"
      >
        <a
          aria-disabled="false"
          class="dusk-anchor dusk-anchor--on-surface dusk-anchor-button dusk-anchor-button--variant--tertiary dusk-anchor-button--size--default dusk-icon-button--labeled receive__button"
          href="/some-base-path/dashboard"
        >
          <svg
            class="dusk-icon dusk-icon--size--default dusk-anchor-button__icon"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
            />
          </svg>
          
           
          <span
            class="dusk-anchor-button__text"
          >
            Back
          </span>
          
          
        </a>
         
        <button
          class="dusk-button dusk-button--type--button dusk-button--variant--primary dusk-button--size--default dusk-icon-button--labeled receive__button"
          type="button"
        >
          <svg
            class="dusk-icon dusk-icon--size--default dusk-button__icon"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
            />
          </svg>
          
           
          <span
            class="dusk-button__text"
          >
            Copy
          </span>
          
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
